p2panda-sync/src/lib.rs

```diff
@@ -1,8 +1,26 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
-#[cfg(feature = "core")]
-mod codec;
-#[cfg(feature = "core")]
-pub mod engine;
 pub mod protocols;
 pub mod traits;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum SyncError {
+    #[error("protocol error: {0}")]
+    Protocol(String),
+    #[error("input/output error: {0}")]
+    IoError(#[from] std::io::Error),
+    #[error("codec error: {0}")]
+    Codec(String),
+    #[error("custom error: {0}")]
+    Custom(String),
+}
```

Comment:
The error types feel a bit indistinct, can it be improved by doc strings or more info in the error strings?

Reply:
I added doc strings, hope that helps.